packages/block-library/src/categories/edit.js

```diff
@@ -203,10 +203,12 @@ class CategoriesEdit extends Component {
 }
 
 export default withSelect( ( select ) => {
-	const { getCategories, isRequestingCategories } = select( 'core' );
+	const { getEntityRecords } = select( 'core' );
+	const { isResolving } = select( 'core/data' );
+	const query = { per_page: -1 };
 
 	return {
-		categories: getCategories(),
-		isRequesting: isRequestingCategories(),
+		categories: getEntityRecords( 'taxonomy', 'category', query ),
+		isRequesting: isResolving( 'core', 'getEntityRecords', 'taxonomy', 'category', query ),
```

Comment:
Is this the correct usage? Specifically the `'core'` argument

https://github.com/WordPress/gutenberg/blob/5632f6880fff49ce8866836db9bba0d1f67cb2d3/packages/core-data/src/selectors.js#L28

Reply:
yes, this is the correct usage because I'm using the root one. The selector in `core/data` not the selector in `core`. I wasn't even aware of the existence of the selector in `core`.

I think I'm against having this selector in `core` unless it's made generic (added automatically to all stores)

Reply:
I think my head is made to spin by the fact that `core/data` is the reducer key for `packages/data` and `core` is the reducer for `packages/core-data` 😖

Reply:
Ok then so isn't it still being used wrongly because `args` is expected as an array, not a spread?

Reply:
Yes, we should probably change that. A graceful transition would be to change `core/data` and alias it with something like `data` (or something else) and deprecated `core/data` and then after sometime do the same thing for `core` (`core-data`)

Reply:
True, it's still wrong, It's not the first time either this difference in passing arguments trick me.